sklearn/linear_model/tests/test_bayes.py

```diff
@@ -77,19 +77,48 @@ def test_prediction_bayesian_ridge_with_constant_input():
 
 def test_std_bayesian_ridge_with_constant_input():
     # Test BayesianRidge standard dev. for edge case of constant target vector
-    # The standard dev. should be relatively small (< 0.01 is tested here)
+    # The standard dev. should be relatively small (< 0.1 is tested here)
     n_samples = 4
     n_features = 5
     constant_value = np.random.rand()
     X = np.random.random((n_samples, n_features))
     y = np.full(n_samples, constant_value)
-    expected_upper_boundary = 0.01
+    expected_upper_boundary = 0.1
 
     clf = BayesianRidge()
     _, y_std = clf.fit(X, y).predict(X, return_std=True)
     assert_array_less(y_std, expected_upper_boundary)
 
 
+def test_prediction_ard_with_constant_input():
+    # Test ARDRegression prediction for edge case of constant target vector
+    n_samples = 4
+    n_features = 5
+    constant_value = np.random.rand()
+    X = np.random.random((n_samples, n_features))
+    y = np.full(n_samples, constant_value)
+    expected = np.full(n_samples, constant_value)
+
+    clf = ARDRegression()
+    y_pred = clf.fit(X, y).predict(X)
+    assert_array_almost_equal(y_pred, expected)
+
+
+def test_std_ard_with_constant_input():
+    # Test ARDRegression standard dev. for edge case of constant target vector
+    # The standard dev. should be relatively small (< 0.1 is tested here)
```

Comment:
That seems quite a large standard dev...

Reply:
tl;dr: I decreased the upper bound to 0.01. 

Playing around with a couple of random initializations, the `std` was around 0.005. To make sure that the test won't fail by chance, I originally chose 0.1 as an upper bound. Now that I use a random state in the test, it's safe to reduce the bound to 0.01.